src/fund/utils/setupEventSubscriptions.ts

```diff
@@ -0,0 +1,42 @@
+import { DEFAULT_ONRAMP_URL } from '../constants';
+import type { EventMetadata } from '../types/events';
+import { subscribeToWindowMessage } from './subscribeToWindowMessage';
+
+type SetupEventSubscriptionsParams = {
+  host?: string;
+  onSuccess?: () => void;
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  onExit?: (error?: any) => void;
+  onEvent?: (event: EventMetadata) => void;
+};
+
+/**
+ * Subscribes to events from the Coinbase Onramp widget.
+ * @param onEvent - Callback for when any event is received.
+ * @param onExit - Callback for when an exit event is received.
+ * @param onSuccess - Callback for when a success event is received.
+ * @returns a function to unsubscribe from the event listener.
+ */
+export function setupEventSubscriptions({
```

Comment:
@rustam-cb I think we should include "onramp" in this function name, since this is specific to onramp events, and the fund component is not necessarily coupled to onramp. Maybe `subscribeToOnrampEvents` or `setupOnrampEventListeners`?